### PR TITLE
ISPN-5727 Stop client listener executor on shutdown

### PR DESCRIPTION
**Pull request by galderz**
* Make sure also that when a client listener is removed, the task is
  cancelled.
* Added a test helper method to detect thread leaks given a regular
  expression.

### CODE REVIEW

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/ClientListenerLeakTest.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+         public void call() {
+            RemoteCache<Integer, String> remote = rcm.getCache();
+            EventLogListener<Integer> eventListener = new EventLogListener<>();
+            remote.addClientListener(eventListener);
+            eventListener.expectNoEvents();
+            remote.put(1, "one");
+            eventListener.expectOnlyCreatedEvent(1, cache());
+         }
+      });
+      detectThreadLeaks(".*Client-Listener.*");
```
Comment (danberindei) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/ClientListenerLeakTest.java]: Won't this fail if there's another test running in parallel and using client listeners?

Comment (danberindei) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/ClientListenerLeakTest.java]: Failed in CI (and on my machine):
http://ci.infinispan.org/viewLog.html?buildId=30019&buildTypeId=bt9&guest=1

**Comment by galderz** [comment]
@danberindei Fixed the issue by adding cache name to thread name and only detecting that thread.

**Comment by danberindei** [comment]
This failure in HotRodOperationsAuthzIT.testSupervisorAddListener seems relevant:
http://ci.infinispan.org/viewLog.html?buildId=30078&buildTypeId=bt9&guest=1

**Comment by danberindei** [comment]
The `HotRodOperationsAuthzIT.testSupervisorAddListener` NPE is still present in the latest build.

**Comment by danberindei** [comment]
Personally I'm not sure the thread pool is needed at all - I would expect most listeners are installed for long periods of time, so there should be no problem with every listener creating its own thread.

**Comment by galderz** [comment]
@danberindei Should be fixed now. Removed the cancelling of individual threads since that's not really needed (fixes the failing unit test too). The important thing is making sure the thread pool is shutdown. I added a little await for termination in the end, for more predictive unit testing.

**Comment by danberindei** [comment]
Integrated, thanks Galder!

**Pull request closed without merging**